src/Controller/EnMarche/DonationController.php

```diff
@@ -63,7 +67,7 @@ public function informationsAction(Request $request, DonationRequestUtils $donat
         }
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $this->get('app.donation_request.handler')->handle($donationRequest);
+            $donationRequestHandler->handle($donationRequest);
```

Comment:
You should keep it public to initialize the service only when the form is submitted, or implement`ServiceSubscriberInterface`

Reply:
I did it but the team make the decision it was overkill

Reply:
Let me clarify.

This optimization is overkill by itself. Set the service public just for that does not worth the shot IMHO. Instantiating one more service on one route every single time is a tiny tiny tiny thing.

`ServiceSubscriberInterface` was used by @cacahouwete but it's and overhead for all the controller while it's only needed here and it makes the code less readable for no good reason.